docs/data-science/data/preparation.md

```diff
@@ -0,0 +1,292 @@
+```
+
+```title=">>> Output"
+Shape of data: (4530, 18); Shape of data_social: (4304, 4)
+```
+
+The output indicates that `data` contains more observations (customers) than
+`data_social`. However, first and foremost it is good practice to check 
+for duplicated data.
+
+```python
+# check for duplicated rows
+print(data.duplicated().sum())
+```
+
+```title=">>> Output"
+np.int64(411)
+```
+
+`data` contains `#!python 411` duplicated rows. These can be removed easily:
+
+```python
+data = data.drop_duplicates()
+```
+
+???+ question "Check for duplicates"
+    
+    Check for duplicates in `data_social` and remove them if necessary.
+
+<?quiz?>
+question: How many duplicates were present in <code>data_social</code>?
+answer: None
+answer: 3760
+answer: 411
+answer-correct: 376
+content:
+<p><code>data_social</code> had 376 duplicated rows.</p>
+<?/quiz?>
+
+???+ info "A note on `#!python pd.DataFrame.drop_duplicates()`"
+        
+    By default, the method `#!python pd.DataFrame.drop_duplicates()` removes
+    all duplicated rows. However, you can pass an argument to `subset` in 
+    order to remove duplicates based on specific columns. For example, if we 
+    want to drop duplicates based on the `id` column, we can do so by:
+    
+    ```python
+    data_social = data_social.drop_duplicates(subset=["id"])
+    ```
+
+    `subset` can also be all list of multiple columns.
+
+## Merge methods
+
+To combine both data sets we will use the `#!python pd.DataFrame.merge()` 
+method to
+
+> Merge DataFrame or named Series objects with a database-style join
+> 
+> -- <cite>[pandas documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html)</cite>
+
+Looking at the `how` parameter we are presented with 5 (!) different options 
+to perform a merge. The most common ones are:
+
+- `#!python "left"`
+- `#!python "right"`
+- `#!python "inner"`
+- `#!python "outer"`
+
+<div style="text-align: center;">
+ <iframe src="https://giphy.com/embed/kaq6GnxDlJaBq" width="218" height="240" style="" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/chloe-concerned-kaq6GnxDlJaBq"></a></p>
+ <figcaption style="text-align: center;">
+      And now what?!
+ </figcaption>
+
+</div>
+
+In order to be able to choose the appropriate method, we need to break them 
+down:
+
+![Merge types](../../assets/data-science/data/merge-types.png)
+
+- **Left join**: The resulting `DataFrame` will contain all rows from the 
+  left `DataFrame` (data 1) and the matched rows from the right `DataFrame` 
+  (data 2).
+- **Right join**: The resulting `DataFrame` will contain all rows from the 
+  right `DataFrame` (data 2) and the matched rows from the left `DataFrame` 
+  (data 1).
+- **Inner join**: The resulting `DataFrame` will contain only the rows that 
+  have matching values in both `DataFrame`s.
+- **Outer join**: The resulting `DataFrame` will contain all rows from both 
+  `DataFrame`s.
+
```

Comment:
I like how you described the different merges. Based on our audience: maybe we can give a small minimal example for each? Or link a website, where a minimal example is shown?

Reply:
I've extended the section on merge methods, to explain provide some additional examples and an 'in-depth' look.